airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -975,6 +976,47 @@ public Optional<ActorCatalogFetchEvent> getMostRecentActorCatalogFetchEventForSo
     return records.stream().findFirst().map(DbConverter::buildActorCatalogFetchEvent);
   }
 
+  @Data
+  public static class ActorCatalogFetchEventWithCreationDate {
+
+    private final ActorCatalogFetchEvent actorCatalogFetchEvent;
+    private final OffsetDateTime createdAt;
+
+  }
+
+  public Map<UUID, ActorCatalogFetchEventWithCreationDate> getMostRecentActorCatalogFetchEventForSources(final List<UUID> sourceIds)
+      throws IOException {
+
+    return database.query(ctx -> ctx.select(ACTOR_CATALOG_FETCH_EVENT.asterisk())
+        .from(ACTOR_CATALOG_FETCH_EVENT)
+        .where(ACTOR_CATALOG_FETCH_EVENT.ACTOR_ID.in(sourceIds))
```

Comment:
this could end up being a lot of records, especially since we'll start doing source refreshes before every sync for every connection. ideally here we would just fetch the most recent record for each source instead of doing that calculation after loading all of them

Reply:
you might need to use a window function for this (I think row_number): https://mode.com/sql-tutorial/sql-window-functions/

Reply:
I tried to do it, the request is fairly complex expecially using JOOQ. I'll have a second try. This table is not the biggest, I will update the select to only keep the fields we are using to avoid any potential size issue.

Reply:
I think it's ok to do this query in a sql string if it's too complicated using jooq

Reply:
Done